zipline/_protocol.pyx

```diff
@@ -76,6 +79,289 @@ cdef class BarData:
+        value is returned.
+
+        If a single asset and a list of fields are passed in, a pandas Series
+        is returned whose indices are the fields, and whose values are scalar
+        values for this asset for each field.
+
+        If a list of assets and a single field are passed in, a pandas Series
+        is returned whose indices are the assets, and whose values are scalar
+        values for each asset for the given field.
+
+        If a list of assets and a list of fields are passed in, a pandas
+        DataFrame is returned, indexed by asset.  The columns are the requested
+        fields, filled with the scalar values for each asset for each field.
+
+        "price" returns the last known close price of the asset.  If there is
+        no last known value, NaN is returned.  If a value is found, and we
+        had to cross an adjustment boundary (split, dividend, etc) to get it,
+        the value is adjusted before being returned.
+
+        "last_traded" returns the date of the last trade event of the asset,
+        even if the asset has stopped trading. If there is no last known value,
+        NaT is returned.
+
+        "volume" returns the trade volume for the current simulation time.  If
+        there is no trade this minute, 0 is returned.
+
+        "open", "high", "low", and "close" return the relevant information for
+        the current trade bar.  If there is no current trade bar, NaN is
+        returned.
+        """
+        if isinstance(assets, Asset):
+            if isinstance(fields, str):
+                # return scalar value
+                return self.data_portal.get_spot_value(
+                    assets,
+                    fields,
+                    self.simulation_dt_func(),
+                    self.data_frequency
+                )
+            else:
+                # assume fields is iterable
+                # return a Series indexed by field
+                return pd.Series(data={
+                    field: self.data_portal.get_spot_value(
+                        assets,
+                        field,
+                        self.simulation_dt_func(),
+                        self.data_frequency)
+                    for field in fields
+                }, index=fields, name=assets.symbol)
+        else:
+            if isinstance(fields, str):
+                # assume assets is iterable
+                # return a Series indexed by asset
+                return pd.Series(data={
+                    asset: self.data_portal.get_spot_value(
+                        asset,
+                        fields,
+                        self.simulation_dt_func(),
+                        self.data_frequency)
+                    for asset in assets
+                    }, index=assets, name=fields)
+            else:
+                # both assets and fields are iterable
+                data = {}
+
+                for field in fields:
+                    series = pd.Series(data={
```

Comment:
I'm open to it, but not optimizing for the multi-asset, multi-field case yet.  can do that later.